libr/anal/fcn.c

```diff
@@ -640,15 +688,25 @@ static int fcn_recurse(RAnal *anal, RAnalFunction *fcn, ut64 addr, ut8 *buf, ut6
 			return R_ANAL_RET_END;
 		case R_ANAL_OP_TYPE_NOP:
 			if (anal->opt.nopskip) {
-				RFlagItem *fi = anal->flb.get_at (anal->flb.f, addr);
-				// do not skip nops if there's a flag at starting address
-				if (!fi || strncmp (fi->name, "sym.", 4)) {
-					if ((addr + delay.un_idx - oplen) == fcn->addr) {
-						fcn->addr += oplen;
-						bb->size -= oplen;
-						bb->addr += oplen;
-						idx = delay.un_idx;
+				if (!strcmp (anal->cur->arch, "mips")) {
+					//Looks like this flags check is useful only for mips
+					// do not skip nops if there's a flag at starting address
+					RFlagItem *fi = anal->flb.get_at (anal->flb.f, addr);
+					if (!fi || strncmp (fi->name, "sym.", 4)) {
+						if ((addr + delay.un_idx - oplen) == fcn->addr) {
+							fcn->addr += oplen;
+							bb->size -= oplen;
+							bb->addr += oplen;
+							idx = delay.un_idx;
+							goto repeat;
+						}
+					}
+				} else {
+					skip_ret = skip_hp (anal, fcn, &op, bb, addr, tmp_buf, oplen, delay.un_idx, &idx);
+					if (skip_ret == 1) {
 						goto repeat;
+					} else if (skip_ret == 2) {
```

Comment:
I need it here, 3rd option is possible when we should just do nothing and continue